### DISCUSSION TITLE
how big can the sensor board flex pcb be?

### DISCUSSION THREAD

**Discussion by rfc6919** [joeycastillo/Sensor-Watch#31] [Ideas]
Is there enough interior space for a flex pcb covering the entire diameter of the main board? I'd be interested replacing the sensor board with an NFC interface. I'm thinking of a circular flex pcb with cutouts for the LCD interface and to make an interior tab for the 9-pin connector, something like a VQFN RF430CL331H where the sensor IC goes, and an antenna around the full circumference. (RF performance is another question, probably comes down to "try it and see")

**Comment by joeycastillo** [comment]
While there's no more room for parts, there may be a bit more space for a PCB feature like an antenna. The L shape is just to get us over to the extra parts area, but you could extend maybe 4 mm further over the microcontroller, into a lumpy sideways T shape? That's about the biggest you could go while still being able to press fit the board in the enclosure:

![DSC_0766 copy 2](https://user-images.githubusercontent.com/1765152/148999119-485c9c4f-701f-444c-888e-c3e66a297247.jpg)

I suspect that you could maybe extend the sensor board PCB out to the full size of the display (~ 11 mm × 21 mm) by removing the white reflective backing sheet from the LCD enclosure, and carefully sliding the sensor board underneath the LCD. No guarantees this would work, and you'd probably want to fill the whole front side of the sensor board with white silkscreen to improve contrast with the LCD: 

![DSC_0766 copy](https://user-images.githubusercontent.com/1765152/148999182-759d2565-60d2-42f0-aae8-b99202f765a3.jpg)

But beyond that, you'll run into plastic bits of the enclosure that will prevent you from going any larger.

**Reply by rfc6919** [reply]
hmmm... cool, well, worth a shot I guess... cheers!

**Comment by bademux** [comment]
NFC will be really nice addition. We can use OTP over nfc to provide codes for Android devices

**Reply by rfc6919** [reply]
I'm not convinced the RF performance will be usable, with a tiny antenna loop sandwiched right between the display and the SAM L22, and the main PCB behind that. assuming the LCD backing sheet is foil, that won't help either (though fully removing the foil to be able to have a slightly larger loop would help there). definitely worth a shot though.

**Reply by joeycastillo** [reply]
For NFC, it would be a more aggressive mod, but the [Data Runner](https://n-o-d-e.net/datarunner.html) by NODE included a PCB frontplate with a larger NFC antenna that would go on the front either in place of or behind the acrylic screen. (I also had it fabricated in OSH Park flex, but I don't have a photo.)

![E9u-p8qXoA8pPnX](https://user-images.githubusercontent.com/1765152/149232283-4dc33cff-2be4-4713-8713-3b16f47b0e4d.jpeg)

There are two contacts on the back of the PCB, and I admit it would be fiddly to wire them to a sensor board, but that's been my thinking for it. Just haven't had a chance to assemble the NFC sensor board or think about coding drivers for it. 

Anyway the gerbers for that antenna design are available for download on [the Data Runner](https://n-o-d-e.net/datarunner.html) page.

**Reply by perigoso** [reply]
NFC would indeed be cool, what if the antenna pcb was a flex pcb and had an fpc like cable coming of it to connect to sensor whatch directly, oh and it would need the nfc chip integrated too

**Comment by stevens-c** [comment]
What size packages can fit on the flex PCB? I've been looking at sensors in LGA & QFN packages and just want to verify that they would work before moving forward.

**Reply by rfc6919** [reply]
per [the docs](https://joeycastillo.github.io/Sensor-Watch-Documentation/),

> the available area for your electronics is quite small (5.7 × 5.7 × 1 mm). Still, this is plenty of room for an environmental sensor, MEMS accelerometer or magnetometer and a couple of decoupling capacitors. Note that you will likely be limited to QFN and LGA type parts; SOICs are too large, and even SSOP packages are generally too thick

**Reply by stevens-c** [reply]
I saw that, however some QFN & LPGA packages claim that with solder their height stands between 1-1.2 mm. Just want to make sure this won't be an issue.

**Reply by joeycastillo** [reply]
I don't have a 100% solid number on this, but I measured from the plastic tabs above the sensor board to the surface of the main PCB, and got 1.1 mm. Factoring in the thickness of the flex PCB, I estimated 1 mm as the maximum. I also tested briefly with a 1mm tall part soldered to a sensor board, and the movement fit together successfully; I didn't see anything on the screen that would indicate a pressure point hitting it.

Having said that I'm just hesitant to say with certainty "this won't be an issue"; the tallest parts I'm using regularly right now are 0.7 and 0.93 mm tall, so I'm not testing this limit with any regularity. (tho the total height of my current accelerometer board is 1.17 mm, which is more than 1.1 mm, so there may be some leeway here?)
